File: app/src/main/java/com/marcinmoskala/arcseekbar/MainActivity.java
package com.marcinmoskala.arcseekbar;

import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ArcSeekBar arcSeekBar = findViewById(R.id.seekArc);

        arcSeekBar.setOnProgressChangedListener(new ProgressListener() {
            @Override
            public void invoke(int progress) {
                Log.i("SeekBar", "Value is " + progress);
            }
        });

        int[] intArray = getResources().getIntArray(R.array.progressGradientColors);
        arcSeekBar.setProgressGradient(intArray);
    }
}

File: app/src/main/java/com/marcinmoskala/arcseekbar/MainActivity.kt
package com.marcinmoskala.arcseekbar

import android.graphics.Color
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        seekArc.onProgressChangedListener = { p ->
            Log.i("SeekBar", "Value is $p")
        }
        val colors = resources.getIntArray(R.array.sliderArc)
        seekArc.setProgressGradient(*colors)
    }

    fun toast(text: String) {
        Toast.makeText(this, text, Toast.LENGTH_SHORT).show()
    }
}
